packages/compass-import-export/src/modules/export.ts

```diff
@@ -0,0 +1,537 @@
+import type { Action, AnyAction, Reducer } from 'redux';
+import { combineReducers } from 'redux';
+import type { ThunkAction, ThunkDispatch } from 'redux-thunk';
+import createLoggerAndTelemetry from '@mongodb-js/compass-logging';
+
+import { gatherFieldsFromQuery } from '../export/gather-fields';
+import type { SchemaPath } from '../export/gather-fields';
+import type { ExportAggregation, ExportQuery } from '../export/export-types';
+import { queryHasProjection } from '../utils/query-has-projection';
+import { globalAppRegistry, dataService } from '../modules/compass';
+
+const { track } = createLoggerAndTelemetry('COMPASS-IMPORT-EXPORT-UI');
+
+export type FieldsToExport = {
+  [fieldId: string]: {
+    path: SchemaPath;
+    selected: boolean;
+  };
+};
+
+// Fields can only be prefixed with one '$'. Otherwise $
+// is not allowed, so this should be a safe separator.
+const FieldSplittingSymbol = '$$$$';
+export function getIdForSchemaPath(schemaPath: SchemaPath) {
+  return schemaPath.join(FieldSplittingSymbol);
+}
+export function getLabelForFieldId(fieldId: string) {
+  return fieldId.split(FieldSplittingSymbol).join('.');
```

Comment:
Why not just store the field paths as arrays? Then you never have to worry about special strings. If you need that as a unique string you can use as a cache key or something you can just JSON.stringify() the array.

Reply:
JSON.stringify works so much better thanks. updated.
I was opting to keep them as maps using the string key because we do a check using the keys to see if the parent of the field is checked here https://github.com/mongodb-js/compass/pull/4262/files#diff-ecb1d137cdb3159f737b523e73fb7fcc1cd0bdd8561e81b23f9378771c7d26bfR222 it would be slower if we have to iterate over an array every time. We could use a tree structure to be more efficient but I think that would introduce more complexity.

Reply:
I think if you want to convert both ways JSON.stringify and JSON.parse makes sense. That's effectively the definition of serialization and what JSON is made for ;) If you only wanted something to use as a key where you don't need to be able to work backwards then a hash is probably more analogous. 

But I think JSON.stringify is probably right here.